### DISCUSSION TITLE
Fisheye, Panorama or Both?

### DISCUSSION THREAD

**Discussion by czimm74** [aaronwmorris/indi-allsky#1105] [Ideas]
Ok, been using the Pi5 setup a few weeks now and it is working a treat.  Seeing how I barely spike the CPU or memory I got to wondering.  I like how the Fisheye to Panorama is, but I would like to have cake and pie together.  Can we configure a setting to be able to do both?  If I am reading the section right, it is one or the other?

**Comment by aaronwmorris** [comment]
I merged #1101 a few days ago and that makes the panoramas a fully supported type with its own database tables and web views.  It is also fully supported in the file transfer functions.

**Reply by czimm74** [reply]
And here I thought I checked in the discussions enough to make sure I wasn't bring up anything old.. I'll update allsky shortly then, testing youtube right now.

**Comment by czimm74** [comment]
Ok, I have updated and tried a night with the Panorama turned on.  It did not generate the kerogram or startrails.  And the movie file can't be played, says it is an unknown mime type.  Wanted to make sure it was not just the update, turned off the panorama and the files all generated correctly last night.

**Reply by aaronwmorris** [reply]
Are you using the in-place processing?

**Comment by czimm74** [comment]
No, I have that setting off.  
I adjusted the image circle down to just slightly bigger than the circle mask (was default last time), left everything else the same to try last night.  It did generate the timelapse, kerogram and a proper movie last night.

The only difference I had between the two would have been the diameter.  So, maybe a user error the first try, improper config.

<img width="718" alt="Screenshot 2024-01-23 at 12 10 44 PM" src="https://github.com/aaronwmorris/indi-allsky/assets/23324301/27ef6f81-cf71-4fbb-aa80-8f44a64c975b">

**Reply by aaronwmorris** [reply]
I think I am going to remove the `In-place processing` option anyway.  I think it will cause problems long term.

If you had that option enabled even for a single image, it might have caused problems with generating the timelapse/keogram/etc.  A single image with different dimensions can cause the entire process to fail.  If this is actually the problem (it might not be), you can go into the image viewer and set the exclude flag on that image (or others) to generate things.

I do not know of anything else that would cause the timelapse generation to fail.  All of my tests are passing as of now.

**Reply by aaronwmorris** [reply]
Merged #1110 to remove in-place processing

**Reply by czimm74** [reply]
Only thing I can think of at the moment was the Diameter was at the default 3000.  So, was double what it needed to be in my case.

I will try running with in-place processing on tonight and see what happens.

**Comment by czimm74** [comment]
Ok, tried the in-place processing last night, and it straight up threw a fit over that.  But unlike before, it did throw me an error message that the timelapse could not be generated last night.

**Reply by aaronwmorris** [reply]
As of #1111 I have updated the startrails and keogram code to only add images that match the dimensions of the first image of the series.  This way, at least an incomplete image will be generated instead of failing completely.  The timelapses may still fail, I will eventually get around to working on the timelapse side of this.

A while back, I modified the code to start recording the image dimensions as columns in the database.  It is very expensive (time/CPU) to open and decode all images to "detect" the dimensions.  I eventually hope to add a more advanced image viewer, and that requires knowing this metadata without actually decoding each file.

**Comment by aaronwmorris** [comment]
Merged #1117 that adds full support for Panorama timelapses.
